sycl/test-e2e/README.md

```diff
@@ -10,19 +11,95 @@
+way as `UNSUPPORTED`. If only one device is matched by `UNSUPPORTED`/`REQUIRES`
+filtering, then `XFAIL` behaves as an actual "expected to fail" marking.
+
+For any device left after the filtering above we expand each `RUN` directive
+containing `%{run}` expansion into one command per device, replacing `%{run}`
+with `env ONEAPI_DEVICE_SELECTOR=<device_matching_requirements> [Optional
+run_launcher if that is configured]` while at the same time properly handling
+`%if` conditions, meaning that the following works as one would expect it to
+behave: `// RUN: %run %t.out %if cpu %{ 1 %} %else %{ 2 %}`.
+`%{run-unfiltered-devices}` is substituted with just `[Optional run_launcher if
+that is configured]`.
+
+Another little nuance is `%{sycl_triple}` substitution. It is constructed by
+concatenating triples for all the devices from `sycl_devices` supported by a given
+test. After that there is also a convenient `%{build}` substitution that is
+equivalent to `%clangxx -fsycl -fsycl-targets=%{sycl_triple} %s`.
 
 # Prerequisites
 
+ - Target runtime(s) to execute tests on devices. See
+   [installation instructions](https://github.com/intel/llvm/blob/sycl/sycl/doc/GetStartedGuide.md#install-low-level-runtime)
  - DPC++ compiler. Can be built following these
    [instructions](https://github.com/intel/llvm/blob/sycl/sycl/doc/GetStartedGuide.md#build-dpc-toolchain)
    or taken prebuilt from [releases](https://github.com/intel/llvm/releases).
  - LIT tools (llvm-lit, llvm-size). They are not available at prebuilts above,
    but can be built in compiler project (e.g. with "ninja check").
- - Target runtime(s) to execute tests on devices. See
-   [installation instructions](https://github.com/intel/llvm/blob/sycl/sycl/doc/GetStartedGuide.md#install-low-level-runtime)
+   
+Last two bullets are only needed for a standalone configuration. During a normal
+development workflow the tests are integrated into a normal project
+build/configuration.
+
+# Developers workflow
+Just build the project according to [Getting Started
+Guide](/sycl/doc/GetStartedGuide.md) and setup your environment per [Run simple
+DPC++ application
+Instructions](/sycl/doc/GetStartedGuide.md#run-simple-dpc-application).
+Then use
+
+```
+# Either absolute or relative path will work.
+llvm-lit <repo_path>/sycl/test-e2e
+```
+
+to run SYCL End-to-End tests on all devices configured in the system. Use
+
+```
+llvm-lit --param sycl_devices="backend0:device0[;backendN:deviceN]*" <repo_path>/sycl/test-e2e
+```
+
+to limit execution to particular devices, where `backend` is one of `opencl`,
+`ext_oneapi_hip`, `ext_oneapi_cuda`, `ext_oneapi_level_zero`,
+`ext_intel_esimd_emulator` and `device` is one of `cpu`, `gpu` or `acc`.
 
-# Build and run tests
+To run individual test use the path to it instead of the top level `test-e2e`
+directory.
+
+# Standalone configuration
+
+This is supposed to be used for CI/automatic testing and isn't recommended for a
+local development setup.
```

Comment:
That sounds strange. I expect my local development setup to match CI/automatic testing. Otherwise, I won't be able to reproduce CI/automatic testing results locally.

Reply:
CI setup is less convenient for developers because it only allows to run tests from its own "build" directory.